packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -156,18 +179,81 @@ class TechnologyController {
 		return response.status(200).send({ success: true });
 	}
 
+	/**
+	 * Delete a technology user.
+	 * DELETE technologies/:idTechnology/users/:idUser
+	 */
+	async deleteTechnologyUser({ params, response }) {
+		const { idTechnology, idUser } = params;
+		let technology;
+		try {
+			technology = await Technology.findOrFail(idTechnology);
+		} catch (error) {
+			throw new ResourceNotFoundException('technology', 400, 'E_RESOURCE_NOT_FOUND');
+		}
+		let user;
+		try {
+			user = await User.findOrFail(idUser);
+		} catch (error) {
+			throw new ResourceNotFoundException('user', 400, 'E_RESOURCE_NOT_FOUND');
+		}
+
+		await technology.users().detach([user.id]);
+
+		return response.status(200).send({ success: true });
+	}
+
 	/**
 	 * Create/save a new technology.
 	 * POST technologies
 	 */
 	async store({ request }) {
 		const data = getFields(request);
 
+		const technology = await Technology.create(data);
+
+		const { users } = request.only(['users']);
+
+		if (users) {
+			for (const user of users) {
+				// eslint-disable-next-line no-await-in-loop
+				await technology.users().attach(user.userId, (row) => {
+					// eslint-disable-next-line no-param-reassign
+					row.role = user.role;
+				});
+			}
+			return Technology.query()
+				.with('users')
+				.where('id', technology.id)
+				.fetch();
+		}
+		return technology;
+	}
+
+	/** POST technologies/:idTechnology/users */
+	async associateTechnologyUser({ params, request }) {
+		const { users } = request.only(['users']);
+		const { idTechnology } = params;
+
+		let technology;
 		try {
-			return Technology.create(data);
+			technology = await Technology.findOrFail(idTechnology);
 		} catch (error) {
-			return error;
+			throw new ResourceNotFoundException('technology', 400, 'E_RESOURCE_NOT_FOUND');
 		}
+
+		for (const user of users) {
+			// eslint-disable-next-line no-await-in-loop
+			await technology.users().attach(user.userId, (row) => {
+				// eslint-disable-next-line no-param-reassign
+				row.role = user.role;
+			});
+		}
+
+		return Technology.query()
+			.with('users')
+			.where('id', technology.id)
+			.fetch();
```

Comment:
não poderia retornar o technology diretamente aqui sem rodar essa query novamente? (a não ser que queira forçar o eager loading antes de retornar)

Reply:
Aqui eu quis usar o eager loading, seria possível retornar apenas o technology.

Reply:
Faz sentido 👌